fenix/app/src/main/java/org/mozilla/fenix/utils/ToolbarPopupWindow.kt

```diff
@@ -82,17 +85,17 @@ object ToolbarPopupWindow {
             }
         }
 
-        clipboardText?.let { text ->
+        if (binding.paste.isVisible) {
             binding.paste.setOnClickListener {
                 popupWindow.dismiss()
-                handlePaste(text)
+                handlePaste(clipboard.text.orEmpty())
             }
         }
 
-        clipboardUrl?.let { url ->
+        if (binding.pasteAndGo.isVisible) {
             binding.pasteAndGo.setOnClickListener {
                 popupWindow.dismiss()
-                handlePasteAndGo(url)
+                handlePasteAndGo(clipboard.extractURL().orEmpty())
```

Comment:
Do we still want to call `handlePasteAndGo` with an empty string? We may want to check for a URL and then we can continue to do the logging `Logger("ToolbarPopupWindow").error("Clipboard contains URL but unable to read text")` if we can't find one

Reply:
My kotlin is a bit rusty, fixed it with a combo of `?.also` and `?:run`.
Could not find something cleaner to unwrap or handle the `null`, unless there is an helper ?